api/src/main/java/com/flipkart/poseidon/oas/SchemaGenerator.java

```diff
@@ -0,0 +1,466 @@
+                            final Content content = new Content();
+                            final MediaType mediaType = new MediaType();
+                            mediaType.schema(createSchema(paramPOJO));
+                            content.addMediaType("application/json", mediaType);
+                            requestBody.content(content);
+                            operation.requestBody(requestBody);
+                            continue;
+                        }
+
+                        operation.addParametersItem(createParameter(paramPOJO).required(true));
+                    }
+                }
+
+                if (paramsPOJO.getOptional() != null) {
+                    for (ParamPOJO paramPOJO : paramsPOJO.getOptional()) {
+                        if (paramPOJO.isFile() || paramPOJO.isBody()) {
+                            continue;
+                        }
+
+                        operation.addParametersItem(createParameter(paramPOJO));
+                    }
+                }
+            }
+
+            ApiResponses responses = new ApiResponses();
+            if (endpointPOJO.getResponse() instanceof String) {
+                String responseString = (String) endpointPOJO.getResponse();
+                responseString = responseString.substring(3, responseString.length() - 2);
+                if (responseString.startsWith("$")) {
+                    responseString = responseString.substring(1);
+                }
+
+                final TaskPOJO taskPOJO = endpointPOJO.getTasks().get(responseString);
+                final Class<? extends DataSource<?>> dsClass = datasources.get(taskPOJO.getName());
+                final Type resolvedType = TypeToken.of(dsClass)
+                        .resolveType(dsClass.getMethod("call").getGenericReturnType())
+                        .getType();
+                resolveAPIResponse("200", resolvedType, responses, modelsDir);
+                handleOtherResponses(dsClass, responses, modelsDir);
+            }
+
+            operation.responses(responses);
+            operation.operationId(endpointPOJO.getName());
+            paths.computeIfAbsent(endpointPOJO.getUrl(), s -> new LinkedHashMap<>()).put(getHttpMethod(endpointPOJO.getHttpMethod()), operation);
+        }
+
+        paths.forEach((k, v) -> {
+            final PathItem pathItem = new PathItem();
+            pathItem.addExtension("x-old-path", k);
+            v.forEach(pathItem::operation);
+            openAPI.path(k, pathItem);
+        });
+
+        final byte[] apiBytes = Json.mapper().writerWithDefaultPrettyPrinter().writeValueAsBytes(openAPI);
+        Files.createDirectories(apiFile.getParent());
+        Files.write(apiFile, apiBytes, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
+
+        modelTypes.forEach((k, v) -> {
+            try {
+                final Path modelFile = modelsDir.resolve(k + ".json");
+                Files.createDirectories(modelFile.getParent());
```

Comment:
Good to delete the existing model files in this directory. In api file case, its ServiceResource.json always, so it gets recreated everytime after truncating old content. But models could have been deleted also.